linera-views/src/key_value_store_view.rs

```diff
@@ -156,18 +157,28 @@ where
     C: Context + Send + Sync,
     ViewError: From<C::Error>,
 {
+    const NUM_INIT_KEYS: usize = 2 + ByteMapView::<C, u32>::NUM_INIT_KEYS;
+
     fn context(&self) -> &C {
         &self.context
     }
 
-    async fn load(context: C) -> Result<Self, ViewError> {
-        let key = context.base_tag(KeyTag::Hash as u8);
-        let hash = context.read_value(&key).await?;
-        let key = context.base_tag(KeyTag::TotalSize as u8);
-        let total_size = context.read_value(&key).await?.unwrap_or_default();
+    fn pre_load(context: &C) -> Result<Vec<Vec<u8>>, ViewError> {
+        let key_hash = context.base_tag(KeyTag::Hash as u8);
+        let key_total_size = context.base_tag(KeyTag::TotalSize as u8);
+        let mut v = vec![key_hash, key_total_size];
         let base_key = context.base_tag(KeyTag::Sizes as u8);
         let context_sizes = context.clone_with_base_key(base_key);
-        let sizes = ByteMapView::load(context_sizes).await?;
+        v.extend(ByteMapView::<C, u32>::pre_load(&context_sizes)?);
+        Ok(v)
+    }
+
+    fn post_load(context: C, values: &[Option<Vec<u8>>]) -> Result<Self, ViewError> {
+        let hash = from_bytes_option(values.first().unwrap())?;
+        let total_size = from_bytes_option_or_default(values.get(1).unwrap())?;
```

Comment:
Not important, but I'd find the following a bit easier to read. (Also in the other files.)

```suggestion
        let hash = from_bytes_option(&values[0])?;
        let total_size = from_bytes_option_or_default(&values[1])?;
```

Alternatively, you could use `values.get(2..).unwrap()` below, if you prefer the explicit `unwrap`s.

Reply:
We should probably raise internal errors instead of panicking.

Reply:
Ok, done.